FortranCon2021-stdlib/stdlib-talk.tex

```diff
@@ -0,0 +1,369 @@
+    This is just scratching the surface; the design is flexible enough to set up fairly sophisticated logging infrastructure.
+    For instance, in a coarray application, one might configure separate loggers for each image or team of images.
+    stdlib now makes it easy to set that up.
+  }
+\end{frame}
+
+
+\begin{frame}[fragile]
+  \frametitle{Demo: stdlib\_bitsets}
+
+  \begin{block}{ex\_bitsets.f90}
+    \small
+\begin{verbatim}
+use stdlib_bitsets
+implicit none
+
+integer :: i; type(bitset_64) :: b1, b2
+
+call b1%from_string('001100') ! S6B001110
+b2 = [(.true., i=1,6)]        ! S6B111111
+
+call xor(b1, b2)              ! S6B110001, S6B111111
+
+call b1%set(2, 4)             ! S6B111111 -- N.B. 0-based index
+print *, b1 == b2             ! T
+end
+\end{verbatim}
+  \end{block}
+
+  \note{
+    The second demo is of the bitsets module.
+    This provides a feature that is often proposed for standardization: an abstract type for working with strings of bits.
+    The basic bitset type uses a 64-bit integer internally, but it keeps track of the actual length of the bitset being represented.
+    Here, I'm initializing two 6-bit sets, one from a string and one from an array of logicals.
+    The comment shows a string representation of the bitset, which can be taken as a bitset 'literal' for the purposes of I/O.
+    Here, the prefix 'S6B' means a set of 6 bits.
+    Most operations with bitsets work in-place rather than creating new instances.
+    For instance, all the usual boolean algebra operations -- like the exclusive-or here -- are implemented as subroutines that overwrite the first argument.
+    Likewise, when one sets or clears single bits or contiguous ranges of bits, the data is modified without making a copy.
+  }
+\end{frame}
+
+
+\begin{frame}
+  \frametitle{New contributors have been key to stdlib's growth}
+
+  \begin{itemize}
+  \item From 16 committors to 34, including our 2 GSoC students
+  \item Over 100 new Issues: bugs, workflow improvements, feature proposals
+  \item From 52 to 97 contributors (commits, discussion, reviews)
+  \end{itemize}
+
+  \begin{block}{Commits to stdlib since FortranCon 2020}
+    \centering
+    \includegraphics[width=\textwidth]{contrib.png}
+  \end{block}
+
+  \note{
+    So we got lots of great new functionality, and in large part that's thanks to an influx of new contributors to the stdlib project.
+    Not only has the number of modules doubles but so has the number of people who've committed changes to stdlib.
+    Included in that are two Google Summer of Code students Aman Godara and Chetan Karwa, who we will hear from shortly.
+    What that figure doesn't include is contributions made in the form of Issues and reviews of Pull Requests.
+    It's really great to see over 100 new Issues opened in the past year, since this not only includes bug reports but also discussion of workflow improvments and proposals for new features.
```

Comment:
```suggestion
    It's really great to see over 100 new Issues opened in the past year, since this not only includes bug reports but also discussion of workflow improvements and proposals for new features.
```